### PR TITLE
Update onechip memory cache readme

### PR DESCRIPTION
**Pull request by jkinsky**
# Existing Sample Changes
## Description

Updated readme to match new template (with exceptions for FPGA), updated branding, updated some formatting, corrected some grammar.

Fixes Issue# 

## External Dependencies

N/A

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [x] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Command Line
- [x] VSCode

### CODE REVIEW

**Comment by jkinsky** [comment]
Adding @jimmytwei for notices.

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
+- How to tune the cache depth
+
+## Prerequisites
+
 | Optimized for                     | Description
 ---                                 |---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
-| Hardware                          | Intel® Agilex™, Arria® 10, and Stratix® 10 FPGAs
+| OS                                | Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| Hardware                          | Intel Agilex® FPGA <br> Intel® Arria® 10 FPGA <br> Intel® Stratix® 10 FPGA
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:24]: I actually prefer it the older way: Intel® Agilex™, Arria® 10, and Stratix® 10 FPGAs
Agilex, Arria 10 and Stratix 10 are FPGA families, so the intention is to say that the sample will work for any of these three family.
(Many different Agilex parts exist within the Agilex FPGA family, etc.)

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
 > When using the hardware compile flow, Intel® Quartus® Prime Pro Edition must be installed and accessible through your PATH.
->
-> :warning: Make sure you add the device files associated with the FPGA that you are targeting to your Intel® Quartus® Prime installation.
 
-## Prerequisites
 
-This sample is part of the FPGA code samples.
-It is categorized as a Tier 3 sample that demonstrates a design pattern.
+> **Warning** Make sure you add the device files associated with the FPGA that you are targeting to your Intel® Quartus® Prime installation.
+
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:38]: Why remove the Prerequisites section?
This section is present in all the other FPGA samples, I can remove it from all of them if you think that this is better this way

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md [hunk 3]
```diff
@@ -100,132 +106,141 @@ This tutorial creates multiple kernels sweeping across different cache depths wi
...
+   ```
+   onchip_memory_cache.fpga_emu.exe
+   ```
+2. Run the sample on the FPGA simulator device.
+   ```
+   set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1
+   onchip_memory_cache.fpga_sim.exe
+   set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=
+   ```
+3. Run the sample on the FPGA device (only if you ran <br> `cmake` with `-DFPGA_DEVICE=<board-support-package>:<board-variant>`).
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:232]: I think that this `<br>` looks weird :(

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md [hunk 4]
```diff
@@ -100,132 +106,141 @@ This tutorial creates multiple kernels sweeping across different cache depths wi
...
+
+1. Run the sample on the FPGA emulator (the kernel executes on the CPU).
+   ```
+   ./onchip_memory_cache.fpga_emu
+   ```
+2. Run the sample on the FPGA simulator device.
+   ```
+   CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1 ./onchip_memory_cache.fpga_sim
+   ```
+3. Run the sample on the FPGA device (only if you ran <br> `cmake` with `-DFPGA_DEVICE=<board-support-package>:<board-variant>`).
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:215]: I think that this <br> looks weird :(

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md (hunk 1 above)
Comment (jkinsky) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:24]: I changed it to match the formulations in the names database. In any event, I'll go with what you want.

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md (hunk 2 above)
Comment (jkinsky) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:38]: I did not remove it. I moved it further up the readme.

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md (hunk 3 above)
Comment (jkinsky) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:232]: Okay. We can remove it.

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md (hunk 4 above)
Comment (jkinsky) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:215]: Okay. We can change it.

**Comment by jkinsky** [comment]
Amended PR to respond to review comments.

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md [hunk 5]
```diff
@@ -1,29 +1,41 @@
...
+- How to tune the cache depth
+
+## Prerequisites
+
 | Optimized for                     | Description
 ---                                 |---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
-| Hardware                          | Intel® Agilex™, Arria® 10, and Stratix® 10 FPGAs
+| OS                                | Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| Hardware                          | Intel® Agilex®, Arria® 10, and Stratix® 10 FPGAs
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:24]:
```suggestion
| Hardware                          | Intel® Agilex™, Arria® 10, and Stratix® 10 FPGAs
```
I think that it's actualy ™ in the database

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md (hunk 2 above)
Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:37]: Oh right, my bad

File: DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md (hunk 5 above)
Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md:24]: Interesting, it seems to have changed recently... I checked in this ™ a few weeks ago, but now it appears to be ®...
I'll make the change to all the other FPGA code samples.

**Review by yuguen-intel — APPROVED** [review]

**Comment by yuguen-intel** [comment]
@ishaghosh27 are you ok with this change?

**Review by ishaghosh27 — APPROVED** [review]

**Pull request merged**